src/server/mod.rs

```diff
@@ -419,7 +423,24 @@ where
                         Command::Pasv => {
                             ensure_authenticated!();
 
-                            let listener = std::net::TcpListener::bind(&passive_addrs.as_slice())?;
+                            let mut rng = rand::thread_rng();
+
+                            let mut listener: Option<std::net::TcpListener> = None;
+                            for _ in 1..BIND_RETRIES {
+                                let i = rng.gen_range(0, passive_addrs.len()-1);
+                                match std::net::TcpListener::bind(passive_addrs[i]) {
+                                    Ok(x) => {
+                                        listener = Some(x);
+                                    }
+                                    Err(_) => continue,
+                                };
+                            };
+
+                            if listener.is_none() {
+                                return Ok(Reply::new(ReplyCode::CantOpenDataConnection, "No data connection established"));
+                            }
+
+                            let listener = listener.unwrap();
```

Comment:
can combine this with the previous if into a match statement that returns on None and sets listener on Some

Something like:
`let listener = match listener {
    None => return Ok(Reply::new(ReplyCode::CantOpenDataConnection, "No data connection established")),
    Some(l) => l.unwrap(),
};`

Reply:
good one